Sources/SpeziHealthCharts/MetricChart/Models/MeasurementCache.swift

```diff
@@ -0,0 +1,79 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import HealthKit
+
+
+/// This functionality will be moved to the `HealthKitDataProvider` implementation.
+actor MeasurementCache {
+    enum CacheError: LocalizedError {
+        case entryNotFound
+        case entryExpired
+        
+        var errorDescription: String? {
+            switch self {
+            case .entryNotFound:
+                String(localized: "No entries found for given key.")
+            case .entryExpired:
+                String(localized: "Entry found in cache has expired.")
+            }
+        }
+    }
+    
+    struct CacheKey: Hashable {
+        let type: HKQuantityType
+        let range: DateInterval
+    }
+    
+    struct CacheValue {
+        let measurements: [HKQuantitySample]
+        let timestamp: Date
+    }
+    
+    private var cache: [CacheKey: CacheValue] = [:]
```

Comment:
You might want to look at [`NSCache`](https://developer.apple.com/documentation/foundation/nscache) as an existing cache implementation that uses cost-based or count-based eviction policies. It seems like you additional want to have a time-based eviction policy, you might be able to build that around the existing NSCache implementation if that is required.